torchgeo/datasets/flair2.py

```diff
@@ -0,0 +1,659 @@
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of datapoints in the dataset.
+
+        Returns:
+            length of dataset
+        """
+        return len(self.files)
+
+    def _load_centroids(self, filename: str) -> dict[str, list[int]]:
+        """Load centroids for mapping sentinel super-areas to aerial patches.
+
+        For detailed information on super-patches, see p.4f of datapaper and  `flair-2_centroids_sp_to_patch.json`.
+        CAUTION: centroids are stored as y, x
+
+        Args:
+            filename: name of the file containing centroids
+
+        Returns:
+            dict: centroids for super-patches
+        """
+        with open(os.path.join(self.root, f"{filename}.json")) as f:
+            centroids = json.load(f)
+            return cast(dict[str, list[int]], centroids)
+    
+    def _get_crop_indices(self, centroid: tuple[int, int]) -> tuple[slice, slice]:
+        """Return indices to crop a super-patch from sentinel data based centroid coordinates.
+        
+        For detailed information on super-patches, see p.4f of datapaper and `flair-2_centroids_sp_to_patch.json`.
+        
+        Args:
+            centroid: centroid coordinates
+
+        Returns:
+            tuple[slice, slice]: crop indices for sentinel data
+        """
+        y, x = centroid
+        eigth_size = self.super_patch_size // 8
+
+        return (slice(x-eigth_size, x+eigth_size), slice(y-eigth_size, y+eigth_size))
+    
+    def _load_files(self) -> list[dict[str, Collection[str]]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, masks
+        """
+        images = sorted(glob.glob(os.path.join(
+            self.root, 
+            self.dir_names[self.split]["images"],
+            "**", self.globs["images"]), recursive=True))
+        
+        sentinels_data = sorted(glob.glob(os.path.join(
+            self.root, 
+            self.dir_names[self.split]["sentinels"],
+            "**", self.globs["sentinels"].format("_data")), recursive=True))
+        sentinels_mask = sorted(glob.glob(os.path.join(
+            self.root, 
+            self.dir_names[self.split]["sentinels"],
+            "**", self.globs["sentinels"].format("_masks")), recursive=True))
+        sentinels = [
+            {"data": data, "snow_cloud_mask": mask}
+            for data, mask in zip(sentinels_data, sentinels_mask)
+        ]
+        
+        masks = sorted(glob.glob(os.path.join(
+            self.root, 
+            self.dir_names[self.split]["masks"],
+            "**", self.globs["masks"]), recursive=True))
+        
+        files = [
+            dict(image=image, sentinel=sentinel, mask=mask)
+            for image, sentinel, mask in zip(images, sentinels, masks)
+        ]
```

Comment:
It seems like this part filters out most of the aerial images. I think it's because each Sentinel-2 image covers multiple aerial images. E.g., `flair_aerial_train/D004_2021/Z1_NN` contains 100 aerial images, and the corresponding `flair_sen_train/D004_2021/Z1_NN` contains one .npy file. So the association between Sentinel-2 and the aerial images should probably be done on a per-folder basis instead of per-file basis. E.g., something like:
```
        sentinel_lookup = {"/".join(s["data"].split("/")[-4:-2]): s for s in sentinels}
        files = [
            dict(image=image,
                 sentinel=sentinel_lookup["/".join(image.split("/")[-4:-2])],
                 mask=mask)
            for image, mask in zip(images, masks)
        ]
```

Reply:
Oh wow! What a severe mistake. Thanks for clarifying. Your approach seems very valid.

Reply:
That happens sometimes :slightly_smiling_face: 

I just tried training a model on the aerial data now with the code posted above, and all seems to be working. I haven't looked closely at the Sentinel-2 data, although they looked correct when I sampled a couple of elements in the `files` list in the debugger.